internal/pkg/repository/repository.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package repository provides support for building and pushing images to a repository.
+package repository
+
+import (
+    "fmt"
+    "github.com/aws/copilot-cli/internal/pkg/aws/ecr"
+)
+
+// DockerService provides support for logging in to repositories, building images and pushing images to repositories.
+type ContainerManager interface {
```

Comment:
Sorry to flip-flop on this, but can we do actually `ContainerLoginBuildPusher`? similar to [io.ReadWriteCloser](https://golang.org/pkg/io/#ReadWriteCloser) (also need to update the comment)